client/app/competitor/page.tsx

```diff
@@ -69,14 +77,14 @@ const EditorButtons = () => {
                     </Button>
                 </Tooltip>
                 <span className="ml-auto">
-                    <Select>
-                        <SelectTrigger className="w-56" defaultValue={currQuestion?.languages?.[0]}>
+                    <Select value={selectedLanguage} onValueChange={handleLanguageSelection}>
```

Comment:
Make this select default to the first language in the list when the user has not yet selected a language.